packages/itwin/tree-widget/src/test/trees/models-tree/ModelsTree.test.tsx

```diff
@@ -252,13 +255,35 @@ describe("ModelsTree", () => {
         }));
       });
 
+      it("should invoke visibilityHandler factory when it is passed as a function", async () => {
+        const node: TreeNodeItem = { id: "test", label: PropertyRecord.fromString("test-node"), isCheckboxVisible: true };
+        setupDataProvider([node]);
+        const visibilityHandlerCallback = () => {return visibilityHandlerMock;};
+        const visibilityHandlerSpy = sinon.spy(visibilityHandlerCallback);
+        const result = render(<ModelsTree {...sizeProps} iModel={imodelMock.object} modelsVisibilityHandler={visibilityHandlerCallback} activeView={mockViewport().object} />);
+        await waitFor(() => result.getByText("test-node"), { container: result.container });
+        expect(visibilityHandlerSpy.calledOnce);
+      });
+
+      it("should dispose of the visibilityHandler when ModelTree component is unmounted", async () => {
+        setupDataProvider([{ id: "test", label: PropertyRecord.fromString("test-node"), isCheckboxVisible: true }]);
+        visibilityHandlerMock.getVisibilityStatus = async () => ({ state: "hidden" });
+        visibilityHandlerMock.dispose = () => {};
+        const disposeSpy = sinon.spy(visibilityHandlerMock, "dispose");
```

Comment:
I think you could actually even do this:
```
visibilityHandlerMock.dispose = sinon.spy();
```
And then
```
expect(visibilityHandlerMock.dispose).to.be.calledOnce;
```

The same can be applied to the tests above where you're stubing and spying on `getVisibilityStatus`.